engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/ExecutionStore.scala

```diff
@@ -102,10 +100,12 @@ final case class ExecutionStore(private val statusStore: Map[JobKey, ExecutionSt
     }
 
     lazy val shardEntriesForCollectorAreDone: Boolean = key match {
-      case collector: CollectorKey => emulateShardEntries(collector).forall(shardKey => doneKeys.exists({ doneKey =>
-        shardKey.scope.fullyQualifiedName == doneKey.scope.fullyQualifiedName &&
-          shardKey.index == doneKey.index
-      }))
+      case collector: CollectorKey =>
+        emulateShardEntries(collector)
+          .map(shard => (shard.scope.fullyQualifiedName, shard.index))
+          .diff(
+            doneKeys.map(key => (key.scope.fullyQualifiedName, key.index))
+          ).isEmpty
```

Comment:
@mcovarr @cjllanwarne I changed this in the last commit which seemed to make it even faster. It makes it go from ~17' to 8' (with job store writing on) and ~11' to ~2' with job writing off.
Just would like a double check that it makes sense and will not break everything.